src/lib.rs

```diff
@@ -1400,17 +1412,17 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 			}
 
 			if let Some(first_route_hop) = path.hops.first() {
-				if let Some(first_hop) = first_hops
-					.iter()
-					.find(|h| h.short_channel_id == Some(first_route_hop.short_channel_id))
+				let path_value = path.final_value_msat() + path.fee_msat();
+				if let Some(remaining_liquidity) =
+					liquidity_map.get_mut(&first_route_hop.short_channel_id)
 				{
-					let path_value = path.final_value_msat() + path.fee_msat();
-					if first_hop.next_outbound_htlc_limit_msat
+					if *remaining_liquidity
 						< path_value * self.config.probing_liquidity_limit_multiplier
```

Comment:
You'll have the same problem as before because the way the division is moved to the other side of the inequality as a multiplication. From the previous example:

`remaining_liquidity`: 100, 81, 62
`path_value * self.config.probing_liquidity_limit_multiplier`: 19 * 3 = 57

Time to write a test? 😅

Reply:
I'm fine with testing upstream. Indifferent on whether you want to land this first.